### PR TITLE
Cst print

### PR DESCRIPTION
**Pull request by clement26695**
Implement the printers for a large part of the input. I leave the rest of the implementation for smaller PRs

### CODE REVIEW

File: packages/prettier-plugin-java/test/hello-world/_output.java [hunk 1]
```diff
@@ -1,5 +1,46 @@
 package graphics.x.y.z;
-public class HelloWorld <AR> extends APZnd implements Leoq, LA {
-  public int t = expression;
-  public void test( int maz, final unannClassType go) throws testException block
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+public class HelloWorld <AR, AKDa> extends APZnd implements Leoq, LA {
```
Comment (bd82) [inline packages/prettier-plugin-java/test/hello-world/_output.java]: We will need to split this to **multiple** smaller test cases

File: packages/prettier-plugin-java/src/printers/types-values-and-variables.js [hunk 2]
```diff
@@ -115,26 +96,53 @@ class TypesValuesAndVariablesPrettierVisitor {
...
+      const startOffset2 = b.name
+        ? b.children.At[0].startOffset
+        : b.startOffset;
+      return startOffset1 - startOffset2;
+    });
+
+    const segments = [];
+    let currentSegment = [];
+
+    for (let i = 0; i < tokens.length; i++) {
```
Comment (bd82) [inline packages/prettier-plugin-java/src/printers/types-values-and-variables.js]: should we use [].forEach or _.forEach

File: packages/prettier-plugin-java/src/printers/types-values-and-variables.js [hunk 3]
```diff
@@ -46,65 +47,45 @@ class TypesValuesAndVariablesPrettierVisitor {
...
+    const tokens = sortClassTypeChildren(
+      ctx.annotation,
+      ctx.typeArguments,
+      ctx.Identifier
+    );
+
+    const segments = [];
+    let currentSegment = [];
+
+    for (let i = 0; i < tokens.length; i++) {
```
Comment (bd82) [inline packages/prettier-plugin-java/src/printers/types-values-and-variables.js]: forEach?

**Comment by bd82** [comment]
please squash this when merging

File: packages/prettier-plugin-java/test/hello-world/_output.java (hunk 1 above)
Comment (clement26695) [reply packages/prettier-plugin-java/test/hello-world/_output.java]: How do you think we should proceed ? One test for each printer function ?

**Comment by bd82** [comment]
Note the travis voter is disabled. somebody with permissions needs to re-enable it 😄

**Pull request merged**